### DISCUSSION TITLE
Newbie question: Cannot connect (from UGS or Arduino IDE serial monitor)

### DISCUSSION THREAD

**Discussion by hmmueller** [Paciente8159/uCNC#555] [General]
I'm a complete newbie to CNC control with Arduino - so if this is not at all the right place to ask, please tell me where I should go. However, also longer googling did not lead me to an answer. My setup is the (I think) default: An Arduino UNO connected to a WIndows 10 laptop, using Arduino IDE (I tried to install PlatformIO, but after it downloaded > 6 GB and still did not work, I gave up on it).

Uploading the standard "Blink" example works, the green LED does its 1 Hz cycles.

What I did: I downloaded µCNC 1.7.2, created config files with the online configurator (cnc_config.h does have a line #define EMULATE_GRBL_STARTUP) and compiled ("Der Sketch verwendet 30282 Bytes (93%) des Programmspeicherplatzes. Das Maximum sind 32256 Bytes.") and uploaded uCNC to the Arduino Uno (which is currently still without the shield - I think this is ok for just connecting to uCNC).

In UGS (64 bit version 2.0.18), I select
- under Firmware "GRBL"
- under Port COM7 (for the USB connection; same one as where I uploaded from Arduino IDE)
- under Baud 115200.
When I now press on the "Connect or Disconnect button", the green LED on the Arduino does a few quick flashes, then another single flash; and I get the following output in the UGS console:

 *** Connecting to jserialcomm://COM7:115200
 >>> ?
 >>> ?
 >>> ?
 **** Connection closed ****

... and that's it. Which obvious action do I overlook?

EDIT: I started Arduino IDE's "Serial Monitor" with "Newline", 115200 Baud. When I type $$<Enter>, I would except the Arduino+µCNC to "do something" (answer with the settings). But nothing is written into the Serial Monitor's output. What do I need to do to get a useful reaction?

EDIT 2: Other examples (e.g. the "CharacterAnalysis" example) work without problems with Serial Monitor; so I assume my cables and hardware are fine. Why's µCNC silent? :-(

EDIT 3: Slowly, I get something up and running. Versions >= 1.7.2 still do not work. But version 1.6.2 does react to Serial Monitor commands; I had to "invert emergency stop" - the documentation has the awkward sentence "and prevent any kind of motion or command appearing to be locked"; actually, it does not "prevent it to be locked", but it "prevents any kind of motion or command".
It is unclear to me why 1.7.2 does not react; and versions in that range require more memory thatn Arduino Uno provides (when using the default configs). But being a newbie, I'll probably be happy with 1.6.2 for a long time :-)!

If anyone can still shed light on why some things die not work --> thanks a lot!!

Harald

**Comment by Paciente8159** [comment]
Hi there,

It would be helpful if you shared you build.json file that with the configuration you used in the web builder to get a sense of what you might have enabled that leads to the code not fit on the UNO.

Now about the behavior of the UNO. I'm assuming you used the UNO environment. The Uno environment comes with control pins enabled and if you let them unconnected the board will remain in ALARM state (because one of the pins is the emergency stop pin).

In this state only a full hand of commands get response out of the UNO. You can confirm this by sending a `?` character in the serial.

If you don't need those pins just disable them by uncommenting [DISABLE_ALL_CONTROLS on cnc_config.h](https://github.com/Paciente8159/uCNC/blob/3518849dead8e4036bbc6de430eef4eb41a7947b/uCNC/cnc_config.h#L470)

I also advise using the v1.7.x-bugfix with all the latest fixes for v1.7.
Version 1.8 is still very big and barely fits UNO and some of the options you need may not allow it to fit the Flash memory.

**Comment by hmmueller** [comment]
Many thanks for quick and detailed answer!

Here is the only *build.json I found (I'm not really sure it's the one I used as build config for 1.7.2, because I desperately tried N versions in M combinations; however, as it's the only one on my computer, it should be the one).

[ucnc_build.json](https://github.com/Paciente8159/uCNC/files/13196730/ucnc_build.json)

Right now, 1.6.2 (which I'd like to use until I understand more ...) reacts to all commands; but UGS 2.0.18 doesn't want to connect - it complains about a missing $22. Might it be that I run into this issue:? https://github.com/winder/Universal-G-Code-Sender/issues/2309 - then, how would I replace the VER string with something UGS accepts?

**Comment by Paciente8159** [comment]
Do you need the control pins (ESTOP, HOLD, etc..) or will you be fully control this via UGS?

**Reply by hmmueller** [reply]
At this time, it's a demo setup - just getting 3 steppers to turn "somehow" ;-). So no, I dont need the control pins - I just disabled them via #define DISABLE_ALL_CONTROLS in cnc-Config.h and uploaded a new version. (Also in the final setup, I intend to run everything with UGS; with Gcode files generated by a special program for my narrow use-cases).

With Arduino's Serial Monitor, I can nicely talk to µCNC. But, see above, UGS disconnects immediately after getting all that initial information - and the "trick" of setting the version string via $I=... apparently does not work with µCNC -> it seems I need to convince µCNC to send a "very GRBL-like VER string": How would I do that?

EDIT: I changed the version string format in protocol.c at line 967, like this:


`//#define VER_INFO __romstr__("[VER: uCNC " CNC_VERSION " - " BOARD_NAME "]" STR_EOL)`
`// HMM: See https://github.com/winder/Universal-G-Code-Sender/issues/2309`
`#define VER_INFO __romstr__("[VER:" CNC_VERSION ":uCNC - " BOARD_NAME "]" STR_EOL)`

And voila - UGS remains connected :-)

**Reply by Paciente8159** [reply]
Try this.
I just downloaded version 1.7.x and disabled control pins and compiled it. Teste with UGS on my PC.
There is something broken on the web builder...must investigate

[firmware.zip](https://github.com/Paciente8159/uCNC/files/13196752/firmware.zip)

**Reply by hmmueller** [reply]
Actually, doesn't work with UGS:
a) I get again the "Reset to continue" messages, which I got before I manually inverted the emergency stop.
b) Version string looks not parsable by UGS, I'd say.

(Actually, I have a third problem - when I connected power to the CNC shield, something smelled ... and no motors are moving. I might have killed something; but the Arduino is still alive ...).

Going to end experiments for now - hae to get up in the morning playing organ at a service ;-)

Thanks a lot!

Harald

**Reply by Paciente8159** [reply]
In that case something might not be right with the board. The disable all controls actually undefines all control pins. That means that even inverting the logic with setting $7 nothing happens.
See my output on a terminal.

- I sent the $7=255 to invert all bits.
- pressed the reset button
- the ? returned Idle state after that

![image](https://github.com/Paciente8159/uCNC/assets/12116536/8a9d5ab9-5874-4ca5-a4ae-bcfadafb261a)

**Comment by Paciente8159** [comment]
Found the issue that I was getting when compiling code for the UNO from the web builder (regarding UART).
I will patch it.
Thanks
